File: server/internal/model/conversation.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/gorm"
	"time"
)

type ConversationType string

const (
	ConversationTypePrivate ConversationType = "private"
	ConversationTypeGroup   ConversationType = "group"
)

type Conversation struct {
	ID            uuid.UUID        `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	Type          ConversationType `gorm:"type:varchar(20);not null" json:"type"`
	Name          string           `gorm:"size:100" json:"name"`
	AvatarURL     string           `gorm:"size:255" json:"avatar_url"`
	OwnerID       uuid.UUID        `gorm:"type:uuid" json:"owner_id"`
	LastMessageAt *time.Time       `json:"last_message_at"`
	CreatedAt     time.Time        `json:"created_at"`
}

func (c *Conversation) BeforeCreate(tx *gorm.DB) error {
	if c.ID == uuid.Nil {
		c.ID = uuid.New()
	}
	return nil
}

type ConversationWithDetails struct {
	ID              uuid.UUID  `json:"id"`
	Type            string     `json:"type"`
	Name            string     `json:"name"`
	AvatarURL       string     `json:"avatar_url"`
	OwnerID         uuid.UUID  `json:"owner_id"`
	LastMessageAt   *time.Time `json:"last_message_at"`
	CreatedAt       time.Time  `json:"created_at"`
	LastMessage     string     `json:"last_message"`
	LastSenderName  string     `json:"last_sender_name"`
	UnreadCount     int64      `json:"unread_count"`
	OtherUserID     uuid.UUID  `json:"other_user_id,omitempty"`
	OtherUserName   string     `json:"other_user_name,omitempty"`
	OtherUserAvatar string     `json:"other_user_avatar,omitempty"`
}

File: client/lib/features/chat/domain/conversation.dart
class Conversation {
  final String id;
  final String type;
  final String? name;
  final String? avatarUrl;
  final String? ownerId;
  final DateTime? lastMessageAt;
  final DateTime? createdAt;
  String? lastMessage;
  String? lastSenderName;
  int unreadCount;
  final String? otherUserId;
  final String? otherUserName;
  final String? otherUserAvatar;
  
  Conversation({
    required this.id,
    required this.type,
    this.name,
    this.avatarUrl,
    this.ownerId,
    this.lastMessageAt,
    this.createdAt,
    this.lastMessage,
    this.lastSenderName,
    this.unreadCount = 0,
    this.otherUserId,
    this.otherUserName,
    this.otherUserAvatar,
  });
  
  factory Conversation.fromJson(Map<String, dynamic> json) {
    return Conversation(
      id: json['id']?.toString() ?? '',
      type: json['type']?.toString() ?? 'private',
      name: json['name']?.toString(),
      avatarUrl: json['avatar_url']?.toString(),
      ownerId: json['owner_id']?.toString(),
      lastMessageAt: json['last_message_at'] != null
          ? DateTime.tryParse(json['last_message_at'])
          : null,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'])
          : null,
      lastMessage: json['last_message']?.toString(),
      lastSenderName: json['last_sender_name']?.toString(),
      unreadCount: json['unread_count'] ?? 0,
      otherUserId: json['other_user_id']?.toString(),
      otherUserName: json['other_user_name']?.toString(),
      otherUserAvatar: json['other_user_avatar']?.toString(),
    );
  }
  
  String get displayName {
    if (type == 'private') {
      return otherUserName ?? name ?? '私聊';
    }
    return name ?? '群聊';
  }
  
  String get displayAvatar {
    if (type == 'private') {
      return otherUserAvatar ?? avatarUrl ?? '';
    }
    return avatarUrl ?? '';
  }
}